### PR TITLE
Is779/paraview persistency

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- [x] add handler to push/pull state using node manager
- [x] automatically restore the state if it is available on start
- [x] automatically save the state on demand through state handler
- [x] added creation of paraview visualizer with NVidia GPU backend
- [x] version 2.11.0

Paraview visualizer in simcore is now using the base paraviewweb image (f.ex: pv-v5.6.0-osmesa/egl) thus not using the integrated kitware wslink launcher as described [here](http://kitware.github.io/paraviewweb/docs/architecture.html#Multi-user-setup) but the simple architecture described [here](http://kitware.github.io/paraviewweb/docs/architecture.html#Simplicity)

Per construction it is not necessary to provide multi-user access as SIM-CORE by default creates a new paraview instance with each node.
This provides now the advantage that refreshing the visualizer will not "lose" the state anymore.

Paraview is using the python twisted webserver, it was modified to allow for using external scripts (in .rpy format). This allows for the creation of the necessary API for retrieving inputs and saving the state of the application.

## Related issue number
links #779

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] **Runs in the swarm**
- [x] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/dy-3dvis/docker-compose.devel.yml [hunk 1]
```diff
@@ -15,21 +15,94 @@ services:
       - POSTGRES_DB=${POSTGRES_DB}
       - POSTGRES_HOST=${POSTGRES_HOST}
       - POSTGRES_PORT=${POSTGRES_PORT}
+      - SIMCORE_NODE_BASEPATH=${SIMCORE_NODE_BASEPATH}
+      - SIMCORE_HOST_NAME=${SIMCORE_HOST_NAME}
       - TEST_DATA_PATH=/home/root/test-data
-      - SERVER_PORT=${SERVER_PORT}
-      - HOST_NAME=${HOST_NAME}
+      - SERVER_PORT=8777
+      - PARAVIEW_DEBUG=0
```
Comment (mguidon) [inline services/dy-3dvis/docker-compose.devel.yml]: PARAVIEW_DEBUG=0 in debug?

File: services/dy-3dvis/docker-compose.yml [hunk 2]
```diff
@@ -1,28 +1,94 @@
-version: '3.4'
+version: '2.3'
```
Comment (mguidon) [inline services/dy-3dvis/docker-compose.yml]: Any reason for this? Do we actually have some kind of rules here?

File: services/dy-3dvis/simcoreparaviewweb/docker/patch_paraview.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+
+# this allow the twisted based server to serve .rpy scripts:
+# https://twisted.readthedocs.io/en/twisted-18.9.0/web/howto/web-in-60/rpy-scripts.html
+sed -i '/^from twisted.web .*/a from twisted.web import script' /opt/paraview/lib/python2.7/site-packages/wslink/server.py
+sed -i 's/^\(.*\)\(root = File(options.content).*$\)/\1root = File(options.content, ignoredExts=(".rpy",))\n\1root.processors = {".rpy": script.ResourceScript}/' /opt/paraview/lib/python2.7/site-packages/wslink/server.py
```
Comment (mguidon) [inline services/dy-3dvis/simcoreparaviewweb/docker/patch_paraview.sh]: Is this a common thing then? I hope this is not deviating to much over time...

File: services/dy-3dvis/docker-compose.devel.yml (hunk 1 above)
Comment (sanderegg) [reply services/dy-3dvis/docker-compose.devel.yml]: this is for debugging paraview webserver itself and is pretty verbose... not necessary to have that one on all the time while doing normal development.

File: services/dy-3dvis/docker-compose.yml (hunk 2 above)
Comment (sanderegg) [reply services/dy-3dvis/docker-compose.yml]: yes, the __runtime__ field is not available in version 3 of compose... necessary to enable the nvidia runtime.

File: services/dy-3dvis/simcoreparaviewweb/docker/patch_paraview.sh (hunk 3 above)
Comment (sanderegg) [reply services/dy-3dvis/simcoreparaviewweb/docker/patch_paraview.sh]: well... common... probably not. it's the only way I found for adding endpoints without changing much more of the code. This part actually allows me to set .rpy files that are recognized on the fly by the paraview webserver and can be used as endpoints (for retrieving the data in the pipeline and so on).
This hopefully would not be necessary once we have a dy-sidecar..

**Pull request merged**